framework/web/Response.php

```diff
@@ -488,6 +485,9 @@ public function sendContentAsFile($content, $attachmentName, $options = [])
             $this->content = $content;
         }
 
+        $mimeType = isset($options['mimeType']) ? $options['mimeType'] : 'application/octet-stream';
+        $this->setDownloadHeaders($attachmentName, $mimeType, !empty($options['inline']), $end - $begin + 1);
```

Comment:
Not really. `$end - $begin + 1` could be more than the length of `$this->content`.